datafusion/expr/src/udf.rs

```diff
@@ -338,6 +346,13 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
     fn monotonicity(&self) -> Result<Option<FuncMonotonicity>> {
         Ok(None)
     }
+
+    // Do the function rewrite.
```

Comment:
```suggestion
    /// Optionally apply per-UDF simplification / rewrite rules
    ///
    /// This can be used to apply function specific simplification rules 
    /// during  optimization (e.g. `arrow_cast` --> `Expr::Cast`).
    ///
    /// Note that since DataFusion handles simplifying arguments 
    /// as well as "constant folding" (replacing a function call with constant
    /// arguments such as `my_add(1,2) --> 3` ) there is no need to implement such
    /// optimizations for UDFs.
```